ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -34,6 +34,42 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
         )
   }
 
+  object CommandErrors extends ErrorGroup() {
+
+    @Explanation("This rejection is given when a command submission could not be processed.")
+    @Resolution("Refer to the error details and/or retry.")
+    object CommandSubmissionFailure
+        extends ErrorCode(
+          id = "COMMAND_SUBMISSION_FAILURE",
+          ErrorCategory.ContentionOnSharedResources,
+        ) {
+      case class Reject(messagePrefix: String, override val throwableO: Option[Throwable])(implicit
+          loggingContext: ContextualizedErrorLogger
+      ) extends LoggingTransactionErrorImpl(
+            cause = {
+              val suffix: String =
+                throwableO.fold("")(t => s": ${t.getClass.getSimpleName}: ${t.getMessage}")
+              s"$messagePrefix$suffix"
+            }
+          )
+    }
+
+    @Explanation("This rejection is given when a command completion has failed.")
+    @Resolution("Refer to the error details and/or retry.")
+    object CommandCompletionFailure
+        extends ErrorCode(
+          id = "COMMAND_COMPLETION_FAILURE",
+          ErrorCategory.ContentionOnSharedResources,
```

Comment:
Indeed this error is used only used where a timeout waiting for completions is reached. But the error description is generic. I would specify that this is a timeout failure (maybe re-use an existing timeout error - I think there is one) to prevent ambiguity and potential misuse.

Reply:
(I think I might have done what you meant here. See thread above (or direct [link](https://github.com/digital-asset/daml/pull/11650/files#r748904886)))